code/game/objects/items/weapons/chemical_flamethrower/chemical_flamethrower.dm

```diff
@@ -0,0 +1,231 @@
+	if(!length(canisters))
+		return
+
+	var/obj/item/chemical_canister/canister_to_remove = canisters[length(canisters)]
+	canister_to_remove.forceMove(get_turf(src))
+	user.put_in_hands(canister_to_remove)
+	canisters -= canister_to_remove
+
+/obj/item/chemical_flamethrower/attackby(obj/item/I, mob/user, params)
+	. = ..()
+	if(!istype(I, /obj/item/chemical_canister))
+		to_chat(user, "<span class='notice'>You can't fit [I] in there!</span>")
+		return
+	if(length(canisters) >= max_canisters)
+		to_chat(user, "<span class='notice'>[src] is already full!</span>")
+		return
+
+	if(!user.unEquip(I))
+		return
+
+	canisters += I
+	I.forceMove(src)
+
+	get_canister_stats()
+
+/obj/item/chemical_flamethrower/proc/get_canister_stats()
+	if(!length(canisters))
+		canister_burn_temp = null
+		canister_burn_duration = null
+		canister_fire_applications = null
+		return
+
+	var/burn_temp
+	var/burn_duration
+	var/fire_applications
+	var/how_many_canisters = length(canisters)
+
+	for(var/obj/item/chemical_canister/canister as anything in canisters)
+		if(!canister.ammo)
+			continue
+		burn_temp += canister.chem_burn_temp
+		burn_duration += canister.chem_burn_duration
+		fire_applications += canister.fire_applications
+
+	canister_burn_temp = round(burn_temp / how_many_canisters, 1)
+	canister_burn_duration = round(burn_duration / how_many_canisters, 1)
+	canister_fire_applications = round(fire_applications / how_many_canisters, 1)
+
+/obj/item/chemical_flamethrower/afterattack(atom/target, mob/user, flag)
+	. = ..()
+	if(flag || !user)
+		return
+
+	if(user.mind?.martial_art?.no_guns)
+		to_chat(user, "<span class='warning'>[user.mind.martial_art.no_guns_message]</span>")
+		return
+
+	if(HAS_TRAIT(user, TRAIT_CHUNKYFINGERS))
+		to_chat(user, "<span class='warning'>Your meaty finger is far too large for the trigger guard!</span>")
+		return
+
+	if(user.get_active_hand() == src) // Make sure our user is still holding us
+		var/turf/target_turf = get_turf(target)
+		if(target_turf)
+			var/turflist = get_line(user, target_turf)
+			add_attack_logs(user, target, "Chemical Flamethrowered at [target.x], [target.y], [target.z]")
+			INVOKE_ASYNC(src, PROC_REF(flame_turf), turflist, user)
+
+/obj/item/chemical_flamethrower/proc/flame_turf(list/turflist = list(), mob/user)
+	if(!length(turflist))
+		return
+
+	var/turf/previousturf = get_turf(src)
+	for(var/turf/simulated/T in turflist)
+		if(!use_ammo(5))
+			to_chat(user, "<span class='warning'>You hear a click!</span>")
+			playsound(user, 'sound/weapons/empty.ogg', 100, TRUE)
+			break // Whoops! No ammo!
+
+		if(T == previousturf)
+			continue	// So we don't burn the tile we be standin on
+
+		var/found_obstruction = FALSE
+		for(var/obj/thing in T)
+			if(thing.density && !istype(thing, /obj/structure/table))
+				found_obstruction = TRUE
+				break
+		if(found_obstruction)
+			break
+
+		make_flame(T)
+		get_canister_stats() // In case we ran out of some fuel this fire
```

Comment:
Might want to rename this proc's name to `update_canister_stats()`

Reply:
Done!